### PR TITLE
Remove ParameterJv

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.14*

## ❌️ General: 91%

<details><summary>45 out of 522 features are not covered by tests</summary><p>

[`ConstantsInBody`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInBody), [`ConstantsInPath`](https://github.com/Azure/autorest.testserver/search?q=ConstantsInPath), [`FileStreamVeryLarge`](https://github.com/Azure/autorest.testserver/search?q=FileStreamVeryLarge), [`FormdataStreamUploadFile`](https://github.com/Azure/autorest.testserver/search?q=FormdataStreamUploadFile), [`HttpRedirect300Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Get), [`HttpRedirect300Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect300Head), [`HttpRedirect301Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Get), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Get), [`HttpRedirect302Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Head), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`HttpRedirect303Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect303Post), [`HttpRedirect307Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Delete), [`HttpRedirect307Get`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Get), [`HttpRedirect307Head`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Head), [`HttpRedirect307Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Patch), [`HttpRedirect307Post`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Post), [`HttpRedirect307Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect307Put), [`OptionalArrayParameter`](https://github.com/Azure/autorest.testserver/search?q=OptionalArrayParameter), [`OptionalClassParameter`](https://github.com/Azure/autorest.testserver/search?q=OptionalClassParameter), [`OptionalImplicitBody`](https://github.com/Azure/autorest.testserver/search?q=OptionalImplicitBody), [`OptionalIntegerParameter`](https://github.com/Azure/autorest.testserver/search?q=OptionalIntegerParameter), [`OptionalStringParameter`](https://github.com/Azure/autorest.testserver/search?q=OptionalStringParameter), [`ResponsesScenarioC200MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC200MatchingModel), [`ResponsesScenarioC201MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC201MatchingModel), [`ResponsesScenarioC404MatchingModel`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioC404MatchingModel), [`UrlPathsArrayCSVInPath`](https://github.com/Azure/autorest.testserver/search?q=UrlPathsArrayCSVInPath), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`getByteEmpty`](https://github.com/Azure/autorest.testserver/search?q=getByteEmpty), [`getByteInvalid`](https://github.com/Azure/autorest.testserver/search?q=getByteInvalid), [`getComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=getComplexPolymorphismComplicated), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`getStringMultiByteCharacters`](https://github.com/Azure/autorest.testserver/search?q=getStringMultiByteCharacters), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`putStringMultiByteCharacters`](https://github.com/Azure/autorest.testserver/search?q=putStringMultiByteCharacters), [`putStringNull`](https://github.com/Azure/autorest.testserver/search?q=putStringNull), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 88%

<details><summary>14 out of 118 features are not covered by tests</summary><p>

[`AzureMethodQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureMethodQueryUrlEncoding), [`AzureODataFilter`](https://github.com/Azure/autorest.testserver/search?q=AzureODataFilter), [`AzurePathQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzurePathQueryUrlEncoding), [`AzureRequestClientIdInError`](https://github.com/Azure/autorest.testserver/search?q=AzureRequestClientIdInError), [`AzureSwaggerQueryUrlEncoding`](https://github.com/Azure/autorest.testserver/search?q=AzureSwaggerQueryUrlEncoding), [`AzureXmsCustomNamedRequestId`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsCustomNamedRequestId), [`AzureXmsCustomNamedRequestIdParameterGroup`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsCustomNamedRequestIdParameterGroup), [`AzureXmsRequestClientIdNull`](https://github.com/Azure/autorest.testserver/search?q=AzureXmsRequestClientIdNull), [`CustomHeaderPostAsyncSucceded`](https://github.com/Azure/autorest.testserver/search?q=CustomHeaderPostAsyncSucceded), [`CustomHeaderPostSucceeded`](https://github.com/Azure/autorest.testserver/search?q=CustomHeaderPostSucceeded), [`CustomHeaderPutAsyncSucceded`](https://github.com/Azure/autorest.testserver/search?q=CustomHeaderPutAsyncSucceded), [`CustomHeaderPutSucceeded`](https://github.com/Azure/autorest.testserver/search?q=CustomHeaderPutSucceeded), [`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson), [`PagingOdataMultiple`](https://github.com/Azure/autorest.testserver/search?q=PagingOdataMultiple)</p></details>

File: src/vanilla/DanModel/DanCodeGenerator.cs [hunk 1]
```diff
@@ -5079,28 +5108,26 @@ internal static string MethodOperationExceptionTypeString(Method method, Setting
...
-                    {
-                        Name = "nextUrl",
-                        SerializedName = "nextUrl",
-                        ModelType = DependencyInjection.New<PrimaryType>(KnownPrimaryType.String),
-                        Documentation = "The URL to get the next page of items.",
-                        Location = ParameterLocation.Path,
-                        IsRequired = true
-                    };
-                    nextUrlParam.Extensions.Add(AzureExtensions.SkipUrlEncodingExtension, true);
+                    Parameter nextUrlParam = DependencyInjection.New<Parameter>();
```
Comment (RikkiGibson) [inline src/vanilla/DanModel/DanCodeGenerator.cs]: What does DependencyInjection do now that we haven't specified any subtype of Parameter to inject? (or maybe I missed something)

File: src/vanilla/DanModel/DanCodeGenerator.cs [hunk 2]
```diff
@@ -6445,5 +6477,250 @@ internal static string IModelTypeDefaultValue(IModelType modelType, Method paren
...
+            => ParameterNeedsConversion(parameter) ? $"{ParameterGetName(parameter).ToCamelCase()}Converted" : ParameterGetName(parameter);
+
+        private static string ParameterConvertToWireType(Parameter parameter, string source, string clientReference)
+        {
+            if (parameter.Location != ParameterLocation.Body &&
+                parameter.Location != ParameterLocation.FormData &&
+                NeedsSpecialSerialization(ParameterGetModelType(parameter)))
+            {
+                IModelType clientType = ParameterClientType(parameter);
+                PrimaryType primary = clientType as PrimaryType;
```
Comment (RikkiGibson) [inline src/vanilla/DanModel/DanCodeGenerator.cs]: No rush but consider changing these to `is PrimaryType primary` when you see them

File: src/vanilla/DanModel/DanCodeGenerator.cs (hunk 1 above)
Comment (daschult) [reply src/vanilla/DanModel/DanCodeGenerator.cs]: It just creates a Parameter object.

**Pull request merged**